doctr/datasets/ic13.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import os
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import numpy as np
+
+from .datasets import AbstractDataset
+
+__all__ = ["IC13"]
+
+
+class IC13(AbstractDataset):
+    """IC13 dataset from `"ICDAR 2013 Robust Reading Competition" <https://rrc.cvc.uab.es/>`_.
+    Example::
+        >>> # NOTE: You need to download both image and label parts from Focused Scene Text challenge Task2.1 2013-2015.
+        >>> from doctr.datasets import IC13
+        >>> train_set = IC13(img_folder="/path/to/Challenge2_Training_Task12_Images",
+        >>>                  label_folder="/path/to/Challenge2_Training_Task1_GT")
+        >>> img, target = train_set[0]
+        >>> test_set = IC13(img_folder="/path/to/Challenge2_Test_Task12_Images",
+        >>>                 label_folder="/path/to/Challenge2_Test_Task1_GT")
+        >>> img, target = test_set[0]
+    Args:
+        img_folder: folder with all the images of the dataset
+        label_folder: folder with all annotation files for the images
+        sample_transforms: composable transformations that will be applied to each image
+        rotated_bbox: whether polygons should be considered as rotated bounding box (instead of straight ones)
+    """
+
+    def __init__(
+        self,
+        img_folder: str,
+        label_folder: str,
+        sample_transforms: Optional[Callable[[Any], Any]] = None,
+        rotated_bbox: bool = False,
+    ) -> None:
+        super().__init__(img_folder)
+        self.sample_transforms = sample_transforms
+
+        # File existence check
+        if not os.path.exists(label_folder) or not os.path.exists(img_folder):
+            raise FileNotFoundError(
+                f"unable to locate {label_folder if not os.path.exists(label_folder) else img_folder}")
+
+        self.data: List[Tuple[Path, Dict[str, Any]]] = []
+        np_dtype = np.float32
+
+        img_names = os.listdir(img_folder)
+
+        for img_name in img_names:
+
+            img_path = Path(img_folder, img_name)
+            label_path = Path(label_folder, "gt_" + Path(img_name).stem + ".txt")
+
+            with open(label_path, "r") as f:
+                _lines = [[x.replace("\"", "").replace(",", "") for x in line.split()] for line in f.readlines()]
```

Comment:
This is the only line that needs extra indentation (we don't need the `f` afterwards)

Reply:
Also to avoid replace (a CSV is actually a text file, the only specificity is the parsing params):
```python
with open("tmp.txt", newline='\n') as f:
    _lines = [
        [val[:-1] if val.endswith(",") else val for val in row]
        for row in csv.reader(f, delimiter=' ', quotechar="'")
    ]
```